test/login.e2e-spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as faker from 'faker';
+import { StatusCodes } from 'http-status-codes';
+import * as request from 'supertest';
+import { factory, useRefreshDatabase, useSeeding } from 'typeorm-seeding';
+import { AppModule } from '../src/app.module';
+import { User } from '../src/entities/user.entity';
+
+describe('A user make a request to the authentication in the system', () => {
+  let app: INestApplication;
+
+  // TODO: change payload type to Partial<LoginDTO>
+  const makePayload = (payload?: { username?: string; password?: string }) =>
+    Object.assign(
+      {
+        username: faker.internet.email(),
+        password: faker.internet.password(),
+      },
+      payload,
+    );
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    await useRefreshDatabase({ configName: 'typeorm-seeding.config.ts' });
+    await useSeeding();
+  });
+
+  it('should return http status code 400 when email or password is bad formated', () =>
+    request(app.getHttpServer())
+      .post('/login')
+      .send({
+        username: '',
+        password: '',
+      })
+      .expect(StatusCodes.BAD_REQUEST));
+
+  it('should return http status code 401 when email is non-registered', () =>
+    request(app.getHttpServer())
+      .post('/login')
+      .send(makePayload())
+      .expect(StatusCodes.UNAUTHORIZED));
+
+  it('should return http status code 401 when email is registered and password is incorrect', async () => {
+    const user = await factory(User)().create();
+    const payload = makePayload({
+      username: user.email,
+      password: faker.internet.password(),
+    });
+
+    await request(app.getHttpServer())
+      .post('/login')
+      .send(payload)
+      .expect(StatusCodes.UNAUTHORIZED);
+  });
+
+  it('should return http status code 200 when valid credentials is given', async () => {
```

Comment:
This test needs to check if jwt is a valid jwt, if expiresIn is a future date. Can you add this checking here?

Reply:
Resolved on 02c383cf2a3d43505b0a17676a4d39605516cfff.